lib/gui/app.js

```diff
@@ -164,25 +189,38 @@ app.run(($window, WarningModalService, ErrorService, FlashStateModel, OSDialogSe
     // Don't open any more popups
     popupExists = true;
 
+    AnalyticsService.logEvent('Close attempt while flashing');
+
     return OSDialogService.showWarning({
       confirmationLabel: 'Yes, quit',
       rejectionLabel: 'Cancel',
       title: 'Are you sure you want to close Etcher?',
       description: messages.warning.exitWhileFlashing()
     }).then((confirmed) => {
       if (confirmed) {
+        AnalyticsService.logEvent('Close confirmed while flashing');
 
         // This circumvents the 'beforeunload' event unlike
         // electron.remote.app.quit() which does not.
         electron.remote.process.exit(EXIT_CODES.SUCCESS);
 
       }
 
+      AnalyticsService.logEvent('Close rejected while flashing');
       popupExists = false;
     }).catch(ErrorService.reportException);
   });
 });
 
+app.run(($rootScope, AnalyticsService) => {
+  $rootScope.$on('$stateChangeSuccess', (event, toState, toParams, fromState) => {
+    AnalyticsService.logEvent('Navigate', {
```

Comment:
I get the following when the application starts:

```
Navigate ({"to":"main","from":""})
```

Maybe worth checking that `from` is defined, and if not, ignore the event?